Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
Just noticed your two previous messages, I must be tired… 

> However, as I said before, I don't really understand what you're trying to do in the case where identifiers are set manually, so perhaps I've missed something?

Some people don't use auto-generated ids, which means simply relying on the presence of an id to determine whether something was persisted or not won't work.

However, using the sample code you provided should work perfectly. Could you please provide a PR for that? I think it is the way to go.